### ISSUE TITLE
Controller name as metric label as option with .UsePrometheusRequestDurations()

### ISSUE THREAD

**Issue by billpieper** [prom-client-net/prom-client-httprequestdurations#44]
We are migrating our applications from Prometheus-net to Prometheus.Client.

Prometheus-net had an extension method to `IApplicationBuilder` called `UseHttpMetrics()`.   With the implementation in Prometheus.Client.HttpRequestDurations, I have been unable to identify a way to incorporate the controller name as a label in the metric.  We really need this as if we use the path instead -- we end up with significant label explosion due to many parameters that are inorporated into our api route paths.

Is there an existing way to get 'Controller' name as a label? and if not, is this something that could be added?

**Comment by sanych-sun** [comment]
Try to use "UseRouteName" it could resolve route name/pattern instead of raw path.

**Comment by phnx47** [comment]
@billpieper "UseRouteName" is working for you?

**Comment by billpieper** [comment]
No, unfortunately this does not meet our needs.  We have [Route()] attributes decorating our controllers at a class level classes like this `[Route("api/[controller]")]`.  Then at methods where we may need to take in a parameter in the route we will have a route attribute defined there too -- something like this:  `[Route("{id}")]`

The `UseRouteName` option ends up setting a label that doesn't let us distinguish between our controllers and routes.

I ended up forking and adding the option to include a label for `Controller` and for `Action`.  This mimics what was available in the old sdk.  Additionally, I added an option to enable the metric to include the Timestamp   Is this something you would consider adding and/or accepting a PR to include?

**Comment by phnx47** [comment]
@billpieper Yes, raise your PR, if work is already done, please. I want to create options to add `Controller` and `Action` to labels, but by default without this, something like this:

```c#
app.UsePrometheusRequestDurations(q =>
{
  q.IncludeConroller = true;
  q.IncludeAction = true;
});
```

**Comment by phnx47** [comment]
Published [v3.5.0](https://www.nuget.org/packages/Prometheus.Client.HttpRequestDurations/3.5.0)
